chaincode/algo.go

```diff
@@ -93,21 +93,32 @@ func queryAlgo(db *LedgerDB, args []string) (out outputAlgo, err error) {
 }
 
 // queryAlgos returns all algos of the ledger
-func queryAlgos(db *LedgerDB, args []string) (outAlgos []outputAlgo, err error) {
+func queryAlgos(db *LedgerDB, args []string) (outAlgos []outputAlgo, bookmark string, err error) {
+	inp := inputBookmark{}
 	outAlgos = []outputAlgo{}
-	if len(args) != 0 {
-		err = errors.BadRequest("incorrect number of arguments, expecting nothing")
+
+	if len(args) > 1 {
+		err = errors.BadRequest("incorrect number of arguments, expecting at most one argument")
 		return
 	}
-	elementsKeys, err := db.GetIndexKeys("algo~owner~key", []string{"algo"})
+
+	if len(args) == 1 && args[0] != "" {
```

Comment:
As it's the first step of full pagination in get/list asset(s) chaincode response, we don't already know if we will send the response to the user or not. Better to make it explicit now I think !

Reply:
I'm not sure I agree on this because for an end user pagination I think you'll change other thing as well and it will not be a wrapped response anymore. But it's not a big issue so you can keep it that way. I just find it less explicit when reading the code.